src/shims/env.rs

```diff
@@ -151,4 +152,29 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         Ok(Scalar::ptr_null(&*this.tcx))
     }
+
+    fn chdir(&mut self, path_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        if !this.machine.communicate {
+            throw_unsup_format!("`chdir` not available when isolation is enabled")
+        }
+
+        let path_bytes = this
+            .memory()
+            .read_c_str(this.read_scalar(path_op)?.not_undef()?)?;
+
+        let path = Path::new(
+            std::str::from_utf8(path_bytes)
+                .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", path_bytes))?,
+        );
```

Comment:
Same concern as in `getcwd`: I think we should go via `OsStr` here, that's the more correct interface than `&str`. Via `OsStrExt`, those can be converted to and from their native raw representation.

Reply:
https://github.com/rust-lang/miri/pull/993